src/apps/openleverage/binance-smart-chain/openleverage.pool.token-fetcher.ts

```diff
@@ -66,12 +80,23 @@ export class BinanceSmartChainOpenleveragePoolTokenFetcher extends AppTokenTempl
     return Number(exchangeRateCurrent) / 10 ** 18;
   }
 
-  async getDataProps({ appToken }: DataPropsStageParams<OpenleverageLpool>) {
+  async getDataProps({ appToken }: DataPropsStageParams<OpenleverageLpool, OpenLeverageDataProps>) {
     const liquidity = appToken.supply * appToken.price;
-    return { liquidity };
+    if (Object.keys(poolDetailMap).length == 0) {
+      const endpoint = `https://bnb.openleverage.finance/api/info/pool/apy`;
```

Comment:
We'd end up calling the APY once for _every_ pool token this way. Instead, create a helper method to get the APY, and annotate it with `@CacheOnInterval`. Then, when you call that method, it'll instead pull from the cache for every pool, rather than hitting the API for every pool.

Reply:
Thank you for your suggestion, please check again when you have time.